.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,46 @@
+name: Nightly integration tests
+
+# Run on request and every day at 3 AM UTC
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 0 3 * * *
+
+jobs:
+  quantinuum:
+    name: Quantinuum (simulator)
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/${{ github.repository }}:latest
+      options: --user root
+
+    steps:
+      - name: Get code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
```

Comment:
Yes, the latest image is generally produced by the latest/current commit on the main branch. 
The sequence of events here is imperfect; it is possible that a commit is on main but the image is not (yet) pushed. If you wanted to guarantee that the test folder matches the image you are pulling, then you could e.g. write the image sha + the github commit sha to a specific file on the bot/data branch, and then at the end of the day, look at the last line in that file, check out these, and delete the rest. If you want, I can take care of that. Let's first get the initial setup going without worrying too much about test and latest image not being in sync.

Reply:
Seems quite involved process. Wouldn't be easier to inject the container a file with the commit SHA used to create it ? We could also output this info to the user on container startup, which would also improve bug reports, since we would be able to ask for the SHA inside the container itself.

Reply:
Good point. This file actually already exists. In the docker container, you can use the command
```
 cat $CUDA_QUANTUM_PATH/assets/build_info.txt | grep -o 'source-sha: \S*' | cut -d ' ' -f 2
```
to get the commit sha to check out on this repo.